packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1647,6 +1653,187 @@ function cleanupParameterData(inputData: NodeParameterValueType): NodeParameterV
 	return inputData;
 }
 
+function findPropertyInArray(
+	name: string,
+	options: INodePropertyMode[],
+): INodePropertyMode | undefined;
+function findPropertyInArray(
+	name: string,
+	options: Array<INodePropertyOptions | INodeProperties | INodePropertyCollection>,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+function findPropertyInArray(
+	name: string,
+	options: Array<
+		INodePropertyOptions | INodeProperties | INodePropertyCollection | INodePropertyMode
+	>,
+):
+	| INodePropertyOptions
+	| INodeProperties
+	| INodePropertyCollection
+	| INodePropertyMode
+	| undefined {
+	const prop = options.find((i) => i.name === name);
+	return prop;
+}
+
+function findPropertyFromParameterName(
+	parameterName: string,
+	nodeType: INodeType,
+): INodePropertyOptions | INodeProperties | INodePropertyCollection {
+	let property: INodePropertyOptions | INodeProperties | INodePropertyCollection | undefined;
+	const paramParts = parameterName.split('.');
+
+	property = findPropertyInArray(paramParts.shift()!, nodeType.description.properties);
```

Comment:
Are you purposefully mutating the paramParts array here? 🙂

Reply:
Yep! It's just so there's less checks in for loop below.